File: core/src/net/natruid/jungle/components/BehaviorComponent.java
package net.natruid.jungle.components;

import com.artemis.Component;
import com.artemis.annotations.PooledWeaver;
import com.badlogic.gdx.utils.IntArray;
import com.badlogic.gdx.utils.ObjectMap;
import net.natruid.jungle.utils.PathNode;
import net.natruid.jungle.utils.ai.BehaviorAction;
import net.natruid.jungle.utils.ai.BehaviorTree;

@PooledWeaver
public class BehaviorComponent extends Component {
    public final IntArray targets = new IntArray();
    public final ObjectMap<Integer, Float> threatMap = new ObjectMap<>();
    public BehaviorTree tree = null;
    public PathNode[] moveArea = null;
    public PathNode[] fullMoveArea = null;
    public BehaviorAction execution = null;
    public float score = 0;

    protected void reset() {
        tree = null;
        moveArea = null;
        fullMoveArea = null;
        execution = null;
        targets.clear();
        threatMap.clear();
    }
}

File: core/src/net/natruid/jungle/components/BehaviorComponent.kt
package net.natruid.jungle.components

import com.artemis.Component
import net.natruid.jungle.utils.Area
import net.natruid.jungle.utils.ai.BehaviorAction
import net.natruid.jungle.utils.ai.BehaviorTree
import java.util.*

class BehaviorComponent : Component() {
    lateinit var tree: BehaviorTree
    val targets = ArrayList<Int>()
    var moveArea: Area? = null
    var fullMoveArea: Area? = null
    val execution = LinkedList<BehaviorAction>()
    var score: Float = 0f
}
